jetty-core/jetty-http/src/main/java/org/eclipse/jetty/http/MimeTypes.java

```diff
@@ -37,6 +39,18 @@
 public class MimeTypes
 {
     static final  Logger LOG = LoggerFactory.getLogger(MimeTypes.class);
+    private static final Set<Locale> KNOWN_LOCALES = Set.copyOf(Arrays.asList(Locale.getAvailableLocales()));
+
+    /**
+     * @param languageTag The language and optional country of the local. For example "en" or "en-GB".
+     * @return A {@link Locale} known to the JVM matching the languageTage or null if the Locale is not known;
+     * @see Locale#forLanguageTag(String)
+     */
+    public static Locale getLocale(String languageTag)
+    {
+        Locale locale =  Locale.forLanguageTag(languageTag);
+        return KNOWN_LOCALES.contains(locale) ? locale : null;
```

Comment:
Well we never handled anything other than language dash country before, so no need to handle every variant.

The cache is not there as a cache, but as a check for known locales.  We use this (and the null return) because the `Accept-Language` header is a quality sorted list.   Previously we have always just taken the first locale from the list and used that even if it is not known to the JVM.    But if we check with known Locales, then we can skip a high q unknown locale in favour of using a lower q value known locale.  Perhaps we should fall back to using the first unknown if there are no known locales?

Note that this was just simple fix of a deprecated locale API in a PR that was otherwise about NPEs.   Perhaps I should revert the Locale stuff to get the NPE fixes in and we can make a Locale specific PR to discuss exactly what we should do with the quality list of Locales?